core/src/main/java/org/apache/iceberg/io/DataWriter.java

```diff
@@ -57,6 +56,12 @@ public DataWriter(FileAppender<T> appender, FileFormat format, String location,
     this.sortOrder = sortOrder;
   }
 
+  @Override
+  public void write(T row) throws IOException {
```

Comment:
Do we need to throw `IOException` if the current writer does not?

Reply:
Probably, no. I guess `IOException` was just defined in the parent interface.